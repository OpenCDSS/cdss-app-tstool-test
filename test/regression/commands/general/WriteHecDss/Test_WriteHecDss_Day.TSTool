# Test setup...
StartLog(LogFile="Results/Test_WriteHecDss_Day.TSTool.log")
# SetDebugLevel(LogFileLevel=10)
# RemoveFile(InputFile="Results\Test_WriteHecDss_Day_out.dss")
RemoveFile(InputFile="Results\Test_WriteHecDss_Day.dssutl.out",IfNotFound=Ignore)
# The generated time series has identifier parts that indicate the HEC-DSS
# parts, other than the D part, which is related to the period of record.
TS Alias1 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1-with-dashes..Cpart1.Day.Fpart1",Description="Test data",SetStart="1950-01-01",SetEnd="1950-03-12",Units="CFS",PatternValues="5,10,12,13,75")
TS Alias2 = NewPatternTimeSeries(NewTSID="Apart2-with-dashes:Bpart2..Cpart2.Day.Fpart2",Description="Test data",SetStart="1950-01-01",SetEnd="1950-03-12",Units="CFS",PatternValues="10,20,24,26,150")
TS Alias3 = NewPatternTimeSeries(NewTSID="No-colon-in-ID..Cpart3.Day.Fpart3",Description="Test data",SetStart="1950-01-01",SetEnd="1950-03-12",Units="CFS",PatternValues="1,2,4,6,15")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_Day_out.dss",Type=PER-AVER)
#
# Use DSSUTL to output the data from the resulting DSS file to compare results
RunDSSUTL(DssFile="Results\Test_WriteHecDss_Day_out.dss",InputFile="Test_WriteHecDss_Day.MCO",DssutlProgram="${WorkingDir}\DSSUTL.EXE")
# In order to regenerate the expected results file, edit the output file in the file: Test_WriteHecDss_Day.MCO
CompareFiles(InputFile1="ExpectedResults\Test_WriteHecDss_Day.dssutl.out",InputFile2="Results/Test_WriteHecDss_Day.dssutl.out",WarnIfDifferent=True)
