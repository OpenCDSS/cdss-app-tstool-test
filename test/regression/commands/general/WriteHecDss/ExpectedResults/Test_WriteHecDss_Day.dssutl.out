  
 //NO-COLON-IN-ID/CPART3/01JAN1950/1DAY/FPART3/
 Start: 01JAN1950 at 2400 hours;   End: 12MAR1950 at 2400 hours;  Number:   71
 Units: CFS        Type: PER-AVER
 01JAN1950, 2400;      1.000     2.000     4.000     6.000    15.000     1.000
 07JAN1950, 2400;      2.000     4.000     6.000    15.000     1.000     2.000
 13JAN1950, 2400;      4.000     6.000    15.000     1.000     2.000     4.000
 19JAN1950, 2400;      6.000    15.000     1.000     2.000     4.000     6.000
 25JAN1950, 2400;     15.000     1.000     2.000     4.000     6.000    15.000
 31JAN1950, 2400;      1.000     2.000     4.000     6.000    15.000     1.000
 06FEB1950, 2400;      2.000     4.000     6.000    15.000     1.000     2.000
 12FEB1950, 2400;      4.000     6.000    15.000     1.000     2.000     4.000
 18FEB1950, 2400;      6.000    15.000     1.000     2.000     4.000     6.000
 24FEB1950, 2400;     15.000     1.000     2.000     4.000     6.000    15.000
 02MAR1950, 2400;      1.000     2.000     4.000     6.000    15.000     1.000
 08MAR1950, 2400;      2.000     4.000     6.000    15.000     1.000
 ----------------------------------------------------------------------------
  
 /APART1/BPART1-WITH-DASHES/CPART1/01JAN1950/1DAY/FPART1/
 Start: 01JAN1950 at 2400 hours;   End: 12MAR1950 at 2400 hours;  Number:   71
 Units: CFS        Type: PER-AVER
 01JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 08JAN1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 15JAN1950, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 22JAN1950, 2400;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 29JAN1950, 2400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 05FEB1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 12FEB1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 19FEB1950, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 26FEB1950, 2400;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 05MAR1950, 2400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 12MAR1950, 2400;     5.00
 ----------------------------------------------------------------------------
  
 /APART2-WITH-DASHES/BPART2/CPART2/01JAN1950/1DAY/FPART2/
 Start: 01JAN1950 at 2400 hours;   End: 12MAR1950 at 2400 hours;  Number:   71
 Units: CFS        Type: PER-AVER
 01JAN1950, 2400;    10.00   20.00   24.00   26.00  150.00   10.00   20.00
 08JAN1950, 2400;    24.00   26.00  150.00   10.00   20.00   24.00   26.00
 15JAN1950, 2400;   150.00   10.00   20.00   24.00   26.00  150.00   10.00
 22JAN1950, 2400;    20.00   24.00   26.00  150.00   10.00   20.00   24.00
 29JAN1950, 2400;    26.00  150.00   10.00   20.00   24.00   26.00  150.00
 05FEB1950, 2400;    10.00   20.00   24.00   26.00  150.00   10.00   20.00
 12FEB1950, 2400;    24.00   26.00  150.00   10.00   20.00   24.00   26.00
 19FEB1950, 2400;   150.00   10.00   20.00   24.00   26.00  150.00   10.00
 26FEB1950, 2400;    20.00   24.00   26.00  150.00   10.00   20.00   24.00
 05MAR1950, 2400;    26.00  150.00   10.00   20.00   24.00   26.00  150.00
 12MAR1950, 2400;    10.00
 ----------------------------------------------------------------------------
  
